src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/components/options_list_popover_suggestions.test.tsx

```diff
@@ -0,0 +1,104 @@
+
+  const allOptions = [
+    { value: 'moo', docCount: 1 },
+    { value: 'tweet', docCount: 2 },
+    { value: 'oink', docCount: 3 },
+    { value: 'bark', docCount: 4 },
+    { value: 'meow', docCount: 5 },
+    { value: 'woof', docCount: 6 },
+    { value: 'roar', docCount: 7 },
+    { value: 'honk', docCount: 8 },
+  ];
+
+  const mountComponent = ({
+    api,
+    displaySettings,
+    stateManager,
+    showOnlySelected,
+  }: {
+    api: any;
+    displaySettings: any;
+    stateManager: any;
+    showOnlySelected?: boolean;
+  }) => {
+    return render(
+      <OptionsListControlContext.Provider
+        value={{
+          api: api as unknown as OptionsListComponentApi,
+          displaySettings,
+          stateManager: stateManager as unknown as ContextStateManager,
+        }}
+      >
+        <OptionsListPopoverSuggestions showOnlySelected={showOnlySelected ?? false} />
+      </OptionsListControlContext.Provider>
+    );
+  };
+
+  test('displays "load more" text when possible', async () => {
+    const mocks = getOptionsListMocks();
+    mocks.api.totalCardinality$.next(allOptions.length);
+    mocks.api.availableOptions$.next(take(allOptions, 5));
+    const suggestionsComponent = mountComponent(mocks);
+
+    // the cardinality is larger than the available options, so display text
+    let optionComponents = await suggestionsComponent.findAllByRole('option');
+    expect(optionComponents.length).toBe(6);
+    expect(
+      suggestionsComponent.queryByTestId('optionsList-control-selection-honk')
+    ).not.toBeInTheDocument();
+    expect(suggestionsComponent.queryByTestId('optionslist--canLoadMore')).toBeInTheDocument();
+
+    // we are displaying all the available options - so don't display "load more" text
+    mocks.api.availableOptions$.next(allOptions);
+    await waitOneTick();
+    optionComponents = await suggestionsComponent.findAllByRole('option');
+    expect(optionComponents.length).toBe(9);
+    expect(
+      suggestionsComponent.queryByTestId('optionsList-control-selection-honk')
+    ).toBeInTheDocument();
+    expect(suggestionsComponent.queryByTestId('optionslist--canLoadMore')).not.toBeInTheDocument();
+  });
+
+  test('only fetch up to maximum request size on scroll', async () => {
+    const mocks = getOptionsListMocks();
+    mocks.api.totalCardinality$.next(100);
+    mocks.api.availableOptions$.next(take(allOptions, 5));
+    const suggestionsComponent = mountComponent(mocks);
+
+    // ensure we fetch the cardinality on scroll
+    expect(mocks.stateManager.requestSize.getValue()).toBe(MIN_OPTIONS_LIST_REQUEST_SIZE);
+    fireEvent.scroll(suggestionsComponent.getByTestId('optionsList--scrollListener'));
+    expect(mocks.stateManager.requestSize.getValue()).toBe(100);
+
+    // reset request size + update cardinality
+    mocks.stateManager.requestSize.next(MIN_OPTIONS_LIST_REQUEST_SIZE);
+    mocks.api.totalCardinality$.next(MAX_OPTIONS_LIST_REQUEST_SIZE + 100);
+    await waitOneTick();
```

Comment:
Same here, instead of `waitOneTick` use `waitFor` and define what you expect the dom to display.

Reply:
Done in 2f853769ada6de3dec591d419aaf5398042232bd :+1: